app/src/main/kotlin/io/homeassistant/companion/android/util/ChangeLog.kt

```diff
@@ -21,15 +42,19 @@ class ChangeLog @Inject constructor(val nightModeManager: NightModeManager) {
             NightModeTheme.DARK -> true
             else -> false
         }
-        if (isDarkTheme) {
-            val darkThemeChangeLog = DarkThemeChangeLog(context)
-            if ((!darkThemeChangeLog.isFirstRunEver && darkThemeChangeLog.isFirstRun) || forceShow) {
-                darkThemeChangeLog.fullLogDialog.show()
-            }
-        } else {
-            val changeLog = ChangeLog(context)
-            if ((!changeLog.isFirstRunEver && changeLog.isFirstRun) || forceShow) {
-                changeLog.fullLogDialog.show()
+
+        // Ensure UI operations happen on Main thread
+        withContext(Dispatchers.Main) {
+            if (isDarkTheme) {
+                val darkThemeChangeLog = createDarkThemeChangeLog(context)
+                if ((!darkThemeChangeLog.isFirstRunEver && darkThemeChangeLog.isFirstRun) || forceShow) {
+                    darkThemeChangeLog.fullLogDialog.show()
+                }
+            } else {
+                val changeLog = createChangeLog(context)
+                if ((!changeLog.isFirstRunEver && changeLog.isFirstRun) || forceShow) {
+                    changeLog.fullLogDialog.show()
+                }
             }
```

Comment:
```suggestion
            val changeLog = if (isDarkTheme) {
                createDarkThemeChangeLog(context)
            } else {
                createChangeLog(context)
            }
            if ((!changeLog.isFirstRunEver && changeLog.isFirstRun) || forceShow) {
                changeLog.fullLogDialog.show()
            }
```

Keep login for first run / force show in one place